### PR TITLE
limit sphinx version

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

**Comment by inducer** [comment]
Is this to address the doc example failure seen by @majosm in #316? Is there a sphinx issue that might explain this?

**Comment by matthiasdiener** [comment]
> Is this to address the doc example failure seen by @majosm in #316? Is there a sphinx issue that might explain this?

Yes, this fixes the doc build at least locally for me (sphinx 7.2.4 is fine, 7.2.5 breaks). There is no issue I could find, but https://github.com/sphinx-doc/sphinx/pull/11645 is the underlying PR that causes the failure.

**Comment by inducer** [comment]
Left a [comment](https://github.com/sphinx-doc/sphinx/issues/11662#issuecomment-1710185052).

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -83,6 +83,11 @@ jobs:
             run: |
                 echo "- matplotlib" >> .test-conda-env-py3.yml
                 curl -L -O https://gitlab.tiker.net/inducer/ci-support/raw/main/ci-support.sh
+
+                # Work around regression introduced in
+                # https://github.com/sphinx-doc/sphinx/pull/11645
```
Comment (inducer) [inline .github/workflows/ci.yml:88]:
```suggestion
                # https://github.com/sphinx-doc/sphinx/pull/11645
                # see also https://github.com/sphinx-doc/sphinx/issues/11662
```

**Comment by alexfikl** [comment]
Unrelated, but why does the documentation take over 1h to generate (almost as much as pylint!)? Is it just that `hello-grudge.py` example?

**Comment by matthiasdiener** [comment]
> Unrelated, but why does the documentation take over 1h to generate (almost as much as pylint!)? Is it just that `hello-grudge.py` example?

Just the `conda env create` takes more than 1 hour for docs and pylint. I suspect that using `mamba env create` would significantly speed this up (at least that's what we saw in mirgecom). Env creation takes more time for these two tests since they pull in matplotlib, which in turn pulls in *many* more packages.

**Comment by alexfikl** [comment]
> Just the `conda env create` takes more than 1 hour for docs and pylint (see e.g. [here](https://github.com/inducer/grudge/commit/37762866f31edd1377e1e334154e2e92b93e35e2/checks/16584867025/logs)). I suspect that using `mamba env create` would significantly speed this up (at least that's what we saw in mirgecom). Env creation takes more time for these two tests since they pull in matplotlib, which in turn pulls in _many_ more packages.

Never thought of looking at that, but yeah, you're completely right! Wow!
Not super familiar with the conda/mamba split, but `ci-support.sh` seems to download `Mambaforge`. Does that not mean it uses `mamba` under the hood?
https://github.com/inducer/ci-support/blob/756aec4768a505b715e3da9ca66d6e72c4f55dfb/ci-support.sh#L116

**Comment by matthiasdiener** [comment]
> Not super familiar with the conda/mamba split, but `ci-support.sh` seems to download `Mambaforge`. Does that not mean it uses `mamba` under the hood? https://github.com/inducer/ci-support/blob/756aec4768a505b715e3da9ca66d6e72c4f55dfb/ci-support.sh#L116

Installing Mambaforge AFAIK only means that `mamba` as a command is available by default, `conda` is still available as a separate command. Currently, ci-support still uses `conda` for env creation: https://github.com/inducer/ci-support/blob/756aec4768a505b715e3da9ca66d6e72c4f55dfb/ci-support.sh#L255 

For some strange reason I'm unable to create a fork on gitlab.tiker.net, so I can't send a PR currently.

**Comment by inducer** [comment]
Thanks!

**Comment by matthiasdiener** [comment]
> Unrelated, but why does the documentation take over 1h to generate (almost as much as pylint!)? Is it just that `hello-grudge.py` example?

With https://github.com/inducer/ci-support/pull/27, the documentation build went from 73 min to [<5 min](https://github.com/inducer/grudge/actions/runs/6113982361/job/16594593960) 👀

**Comment by alexfikl** [comment]
> With [inducer/ci-support#27](https://github.com/inducer/ci-support/pull/27), the documentation build went from 73 min to [<5 min](https://github.com/inducer/grudge/actions/runs/6113982361/job/16594593960) 👀

Much more environmentally friendly! Thanks for figuring this out! :heart:

**Pull request merged**